content/advent-2016/peachpy.md

```diff
@@ -0,0 +1,377 @@
++++
+linktitle = "PeachPy"
+date = "2016-12-02T11:49:46+01:00"
+author = [ "Damian Gryski" ]
+title = "Writing Go assembly functions with PeachPy"
+series = ["Advent 2016"]
+draft = true
++++
+
+**What is PeachPy**
+
+[PeachPy](https://github.com/Maratyszcza/PeachPy) is a Python-based framework
+for writing modules in assembly.  It automates away some of the details and
+allows you to use Python to generate repetitive assembly code sequences.
+
+PeachPy supports writing modules that you can use directly from Go for x86-64.
+(It also supports NaCl and syso modules, but I won't cover those in this post.)
```

Comment:
Actually, talking a bit about SysO would be good. Some features (e.g. using in-memory constants) are not supported in Golang listing output, but work with SysO output.